packages/downloads/src/portal/portal-request-download-metadata.ts

```diff
@@ -134,26 +144,44 @@ function extractLastEditDate(layers: ILayerDefinition[]) {
 }
 
 function formatDownloadMetadata(params: any) {
-  const { cachedDownload, serviceLastEditDate, authentication } = params;
+  const {
+    cachedDownload,
+    serviceLastEditDate,
+    authentication,
+    target
+  } = params;
 
   const lastEditDate =
     serviceLastEditDate !== undefined
       ? new Date(serviceLastEditDate).toISOString()
       : undefined;
 
+  const recentlyUpdated = isRecentlyUpdated(target, serviceLastEditDate);
+
   if (!cachedDownload) {
     return {
       downloadId: composeDownloadId(params),
-      status: "not_ready",
+      status: recentlyUpdated ? "locked" : "not_ready",
       lastEditDate
     };
   }
 
   const { created, id } = cachedDownload;
+
+  const outOfDate = serviceLastEditDate && serviceLastEditDate > created;
+
+  let status;
+  if (recentlyUpdated && outOfDate) {
+    status = "stale_locked";
+  } else if (outOfDate) {
+    status = "stale";
+  } else {
+    status = "ready";
+  }
+
```

Comment:
I'd move the determination of status to it's own function, will make this more concise and easier to read.

Reply:
Good call. This is done.